Common/Data/Auxiliary/FactorFile.cs

```diff
@@ -49,6 +49,12 @@ public class FactorFile : IEnumerable<FactorFileRow>
         /// </remarks>
         public DateTime? FactorFileMinimumDate { get; set; }
 
+        /// <summary>
+        /// Gets the most recent factor change in the factor file
+        /// </summary>
+        public DateTime MostRecentFactorChange => SortedFactorFileData.Reverse()
+            .FirstOrDefault(kvp => kvp.Key != Time.EndOfTime).Key;
```

Comment:
Will (Default).Key be null exception?

Reply:
`KeyValuePair<TKey, TValue>` is a value type, so no null possible, in the case where's `SoratedFactFileData` is empty, this will return `default(DateTime)`